keep-runtime/src/workload.rs

```diff
@@ -32,25 +32,35 @@ pub fn run<T: AsRef<[u8]>, U: AsRef<[u8]>, V: std::borrow::Borrow<(U, U)>>(
     args: impl IntoIterator<Item = T>,
     envs: impl IntoIterator<Item = V>,
 ) -> Result<Box<[wasmtime::Val]>> {
-    let engine = wasmtime::Engine::default();
+    let mut config = wasmtime::Config::new();
+    // Prefer dynamic memory allocation style over static memory
+    config.static_memory_maximum_size(0);
+    let engine = wasmtime::Engine::new(&config);
+
     let store = wasmtime::Store::new(&engine);
 
     // Instantiate WASI
     let mut builder = wasi_common::WasiCtxBuilder::new();
     builder.args(args).envs(envs);
     let ctx = builder.build().or(Err(Error::RuntimeError))?;
-    let wasi_snapshot_preview1 = wasmtime_wasi::Wasi::new(&store, ctx);
+
+    let runtime_payload = wasmtime_wasi::Wasi::new(&store, ctx);
 
     let instance = {
-        let module = wasmtime::Module::new(&store, bytes).or(Err(Error::RuntimeError))?;
+        let module = wasmtime::Module::from_binary(&store.engine(), bytes.as_ref())
+            .or(Err(Error::RuntimeError))?;
+
         let imports = module
             .imports()
             .map(|import| {
                 let module_name = import.module();
                 let field_name = import.name();
+
+                //TODO - more intelligent checking probably useful here
                 let export = match module_name {
-                    "wasi_snapshot_preview1" => Ok(wasi_snapshot_preview1.get_export(field_name)),
-                    _ => Err(Error::ImportModuleNotFound(module_name.to_string())),
+                    "" => Err(Error::ImportModuleNotFound(module_name.to_string())),
+
+                    _ => Ok(runtime_payload.get_export(field_name)),
```

Comment:
The logic doesn't seem to be correct. This part is to resolve WASI symbols from the current WASI snapshot named "wasi_snapshot_preview1". If the same symbol is found in other modules (caught by _), it may not work.

Ideally we should update this part using the new [Linker](https://docs.rs/wasmtime/0.19.0/wasmtime/struct.Linker.html) interface, but in the meantime I'd suggest rather not touching that part.

Reply:
OK - I didn't honestly know what I was doing here.  As we don't know ahead of time what the WASI file that we receive will be, we can't check against anything known.  This was honestly a quick hack to get around this.  Looks like I need to check the Linker interface.

Reply:
I've opened a PR for this as #806.

Reply:
You beat me to it!

Reply:
I believe this is now fixed: I rebased on the the merge version with Daiki's PR included.